SU2_CFD/src/solvers/CGradientSmoothingSolver.cpp

```diff
@@ -0,0 +1,1011 @@
+
+  if (element_container != nullptr) {
+    for (unsigned int iVar = 0; iVar < MAX_TERMS; iVar++) {
+      for (unsigned int jVar = 0; jVar < MAX_FE_KINDS; jVar++) {
+        delete element_container[iVar][jVar];
+      }
+      delete [] element_container[iVar];
+    }
+    delete [] element_container;
+  }
+
+  delete nodes;
+
+}
+
+void CGradientSmoothingSolver::ApplyGradientSmoothingVolume(CGeometry* geometry, CNumerics* numerics,
+                                                            const CConfig* config) {
+  /*--- current dimension if we run consecutive on each dimension ---*/
+  unsigned int iDim = 0;
+
+  /*--- Set vector and sparse matrix to 0 ---*/
+  LinSysSol.SetValZero();
+  LinSysRes.SetValZero();
+  Jacobian.SetValZero();
+
+  /*--- Compute the stiffness matrix for the smoothing operator. ---*/
+  Compute_StiffMatrix(geometry, numerics, config);
+
+  /*--- Impose boundary conditions to the RHS and solve the system. ---*/
+  if (config->GetSmoothSepDim()) {
+    for (iDim = 0; iDim < nDim ; iDim++) {
+
+      SetCurrentDim(iDim);
+
+      Compute_Residual(geometry, config);
+
+      Impose_BC(geometry, config);
+
+      Solve_Linear_System(geometry, config);
+
+      WriteSensitivity(geometry, config);
+
+      LinSysSol.SetValZero();
+      LinSysRes.SetValZero();
+    }
+
+  } else {
+    Compute_Residual(geometry, config);
+
+    Impose_BC(geometry, config);
+
+    Solve_Linear_System(geometry, config);
+
+    WriteSensitivity(geometry, config);
+  }
+}
+
+void CGradientSmoothingSolver::ApplyGradientSmoothingSurface(CGeometry* geometry, CNumerics* numerics,
+                                                             const CConfig* config) {
+  /*--- Set vector and sparse matrix to 0 ---*/
+  LinSysSol.SetValZero();
+  LinSysRes.SetValZero();
+  Jacobian.SetValZero();
+  std::fill(visited.begin(), visited.end(), false);
+
+  /*--- Loop over all DV markers to compute the stiffness matrix for the smoothing operator. ---*/
+  for (unsigned short iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+    if (config->GetMarker_All_DV(iMarker) == YES) {
+      /*--- Compute the stiffness matrix for the smoothing operator. ---*/
+      Compute_Surface_StiffMatrix(geometry, numerics, config, iMarker);
+
+      Compute_Surface_Residual(geometry, config, iMarker);
+
+      if (config->GetDirichletSurfaceBound()) {
+        BC_Surface_Dirichlet(geometry, config, iMarker);
+      }
+    }
+  }
+
+  /*--- Set the matrix to identity if the current mpi rank holds no part of the DV marker. ---*/
+  Complete_Surface_StiffMatrix(geometry);
+
+  /*--- Solve the system and write the result back. ---*/
+  Solve_Linear_System(geometry, config);
+
+  WriteSensitivity(geometry, config);
+}
+
+void CGradientSmoothingSolver::ApplyGradientSmoothingDV(CGeometry *geometry, CNumerics *numerics, CSurfaceMovement *surface_movement, CVolumetricMovement *grid_movement, CConfig *config, su2double** Gradient) {
```

Comment:
Did you run clang-format on new files? This line looks long.

Reply:
I used git clang-format.
Strange that it did not pick up this line, but I changed it now.